### PR TITLE
created a new weather app that generates token_id's which expire every one hour

### PR DESCRIPTION
**Pull request by sravs47**

### CODE REVIEW

File: weather/api/token.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class Token(Resource):
+    def get(self,token_id):
+        pass
+
+class Token_transactions(Resource):
+    def post(self):
+        data = request.json
+        to_model = customer(data['username'],data['password'])
+        db_time = tokens_midlware().get_time(to_model)
+        if is_newerthan_1hour(db_time):
```
Comment (clipod) [inline weather/api/token.py]: you cannot write business logic in resource layer. Create a module in middleware

File: weather/middleware/tokens_middleware.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+
+from weather.database_entities.customer_entity import customer
+from weather.database_model import db
+import uuid
+import datetime
+
+class tokens_midlware:
```
Comment (clipod) [inline weather/middleware/tokens_middleware.py]: check spelling for middleware

File: weather/middleware/tokens_middleware.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        if cust_data is not None:
+            cust_data.username = customer_model.username
+            cust_data.password = customer_model.password
+            cust_data.token_id = uuid.uuid4()
+            cust_data.time = datetime.datetime.now()
+            db.session.add(cust_data)
+            db.session.commit()
+        return cust_data.token_id
+
+    def get_time(self,customer_model):
```
Comment (clipod) [inline weather/middleware/tokens_middleware.py]: good way of breaking down the functionality

File: weather/utils.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+def is_newerthan_1hour(time):
+    current_time =datetime.datetime.now()
+    total_difference =int(timedelta.total_seconds(current_time-time))
+    if total_difference>3600:
+        return False
+    return True
+
+
+JSON_MIME_TYPE = 'application/json'
```
Comment (clipod) [inline weather/utils.py]: you need to find a better place

File: weather/views.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+from weather import app
+
+@app.route("/")
```
Comment (clipod) [inline weather/views.py]: remove this crap

File: weather/middleware/tokens_middleware.py (hunk 2 above)
Comment (sravs47) [reply weather/middleware/tokens_middleware.py]: I did not want to keep module name and class name same, So changed a lil bit.

**Pull request merged**